frontend/pages/index.js

```diff
@@ -1,13 +1,25 @@
-import Header from '../components/Header';
-import RunList from '../components/RunList';
+import Header from "../components/Header";
+import RunList from "../components/RunList";
+import { useEffect, useState } from "react";
+import RunStoreService from "../services/runStore";
 
 const Home = () => {
+    const [isLoading, setIsLoading] = useState(true);
+
+    useEffect(() => {
+        const execService = async () => {
+            await RunStoreService.fetchRunData();
+            setIsLoading(false);
+        }
+        execService();
+    });
```

Comment:
Does this not work?

```javascript
    useEffect(async () => {
        await RunStoreService.fetchRunData();
        setIsLoading(false);
    });
```

Reply:
Nope this is not recommended by React, `useEffect` should be synchronous

Reply:
Do we need to fetch the data every time the component updates? Can we not prevent the calls in multiple re-renders by skipping effects?

https://reactjs.org/docs/hooks-effect.html#tip-optimizing-performance-by-skipping-effects